Import jest structure from lisk modular - Closes #2838 (#2934)
LiskHQ/lisk: README.md

  * [Examples](#examples)
* [Tests](#tests)

>   * [Preparing Node](#preparing-node)

Reviewer: Please add some description here or in sub-headings that which part of the code is covered with mocha and which is covered with jest. And how we are merging the coverage of both to show in the repository tags.

Author: @nazarhussain I added the description. Merging coverage results will be handled in this issue: https://github.com/LiskHQ/lisk/issues/2946

### Running Tests

> Starting from version `1.6.0`, Lisk will be using [Jest](https://jestjs.io) as its main test runner and gradually deprecate [mocha](https://mochajs.org). Since, rewriting all existing mocha tests is not feasible at the moment, we are going to have 2 test runners in our code base.

Reviewer 2: ```suggestion
Starting from version `1.6.0`, Lisk Core will be using [Jest](https://jestjs.io) as its main test runner and gradually deprecate [mocha](https://mochajs.org). Since rewriting all existing mocha tests is not feasible at the moment, we are going to have two test runners in our code base:
```

> 

Reviewer 2: ```suggestion
```

> Until we deprecate mocha completely,

Reviewer 2: ```suggestion
```

> * All source code under `framework/src/modules` folder will be tested by `mocha` and test files should be located under `framework/test/mocha`.

Reviewer 2: ```suggestion
* Modules (all source code under `framework/src/modules` folder) will be tested using `mocha` and test files should be located under `framework/test/mocha`.
```

> * Rest of the source files will be tested by `Jest` and test files should be located under `framework/test/jest`.

Reviewer 2: ```suggestion
* Framework (all of the source files but `framework/src/modules`) will be tested using `jest` and test files should be located under `framework/test/jest`.
* Functional and Network tests suites will be using `mocha` and test files should be located under `framework/test/mocha`.
```

npm run mocha -- path/to/test.js
```

#### v1.6.0 Update

> Starting from version `1.6.0`, Lisk will be using [Jest](https://jestjs.io) as its main test runner and gradually deprecate [mocha](https://mochajs.org). Since, rewriting all existing mocha tests is not feasible at the moment, we are going to have 2 test runners in our code base until we refactor modules and move them to lisk-elements repository.

Reviewer 3: In order to be the main test runner, Jest must cover most of the test. We shouldn't state this at this point because it's simply not true.

Author: > Lisk **will be** using

That's why I stated that from now on it will be the main test runner. Having legacy tests in mocha doesn't conflict with this.

Reviewer: > until we refactor modules and move them to lisk-elements repository

I don't think that's true. Lisk Elements is libraries collection not the integration of libraries. Unless we are 100% sure about the direction, we should not mention such in readme. Below statement is good enough for now: 

> Since, rewriting all existing mocha tests is not feasible at the moment, we are going to have 2 test runners in our code base. Later we plan for further steps.

npm run mocha -- path/to/test.js
```

#### Running Jest Tests

> 

Author: @Tschakki https://github.com/LiskHQ/lisk/issues/2839 after completing this issue, we'll have tests in place with 1.6.0, so I think we can merge it like this at the moment.
